end_of_line.cc

```diff
@@ -0,0 +1,67 @@
+# include <iostream>
+# include <string> 
+# include <algorithm>
+# include <vector>
+# include <numeric>
+
+
+void print(std::vector<int> const &v) {
+		for ( int c: v) { std::cout << c << " "; }
+		
+		std::cout << std::endl;
+}
+
+int main() {
+	
+	auto xor_string = [](std::string &s, int num) -> void { 
+		//std::transform(s.begin(), s.end(), s.begin(),
+		//	[=](char &c) -> char { return c ^ num; });
+		for (char &c: s) { c = c ^ num; }
```

Comment:
the commented one is correct.
beside that you got everything by value: `=`. you dont need it